File: part-two-typescript/src/manifests.ts
import { Dashboard } from "@grafana/grafana-foundation-sdk/dashboard";

export interface Manifest {
    apiVersion: string;
    kind: string;
    metadata: Record<string, string>;
    spec: any;
}

export const dashboardManifest = (folderUid: string, dashboard: Dashboard): Manifest => {
    return {
        apiVersion: 'grizzly.grafana.com/v1alpha1',
        kind: 'Dashboard',
        metadata: {
            folder: folderUid,
            name: dashboard.uid!,
        },
        spec: dashboard,
    };
};

File: part-one-python-starter/src/manifests.py
from grafana_foundation_sdk.models import dashboard, resource


class Manifest:
    @classmethod
    def dashboard(cls, folder_uid: str, dash: dashboard.Dashboard) -> resource.Manifest:
        return resource.Manifest(
            api_version="dashboard.grafana.app/v1alpha1",
            kind="Dashboard",
            metadata=resource.Metadata(
                annotations={"grafana.app/folder": folder_uid},
                name=dash.uid,
            ),
            spec=dash,
        )

